src/Database/Type/FloatType.php

```diff
@@ -65,18 +65,18 @@ public function __construct($name = null)
      *
      * @param string|resource $value The value to convert.
      * @param \Cake\Database\Driver $driver The driver instance to convert with.
-     * @return string|resource
+     * @return string|null
      */
     public function toDatabase($value, Driver $driver)
     {
         if ($value === null || $value === '') {
             return null;
         }
         if (is_array($value)) {
-            return 1;
+            return '1';
         }
 
-        return (float)$value;
+        return sprintf('%F', (float)$value);
```

Comment:
OK. Now I think it would be better to keep as it is, at least regarding FloatType class.

What about adding a note to [the Data Types section of 3.x Cookbook](http://book.cakephp.org/3.0/en/orm/database-basics.html#data-types)? As of 3.x, this kind of issue could happen not only float, but also integer. `int unsigned` doesn't work as expected, if a number is greater than 2147483647 on 32-bit php. `bigint unsigned` never works on any platforms. If we want to use such data types, we need to map the field to `string` always. I think using `numeric(20, 0)` is the same situation.